lib/image_sizes.rb

```diff
@@ -25,12 +27,20 @@ def cache
     end
 
     def cache_image_size(src)
-      @@cache[src] ||= FastImage.size(src, raise_on_failure: false, timeout: timeout)
+      # As src may be the compiled asset path or the local file
```

Comment:
It's just the sizes; we have a `images.yml` file for other attributes such as the `alt` text but it felt like hard-coding the image sizes was riskier as it can impact how they display (if an image was replaced but the size not updated it could cause it to skew).

Reply:
For context; we cache the image sizes for an optimisation targeting Content Layout Shift - if the browser doesn't know the size of an image when it parses the DOM it can't reserve the correct amount of space and a 'shift' happens when the image is finally downloaded and fitted into the document. This library ensures all our `img` tags have explicit `width` and `height` attributes so the browser can immediately calculate the size to reserve (weighing in the CSS applied to the element).

Reply:
That's clear. Thanks for the explanation.